### ISSUE TITLE
Calling Navigator's push() with rootNavigator=true causes keyboard to stop working inside CupertinoTabScaffold

### ISSUE THREAD

**Issue by colehorvitz** [flutter/flutter#99266]
Calling `Navigator.of(context, rootNavigator=true).push(...)` from within a `CupertinoTabScaffold` causes issues with text inputs. Specifically, after calling `push` with `rootNavigator=true`, the keyboard no longer shows up when interacting with text inputs inside the `CupertinoTabScaffold`.

In my app, I use a `CupertinoTabScaffold` with `CupertinoTabView`s as children. There is a use case in which I would like to push a route using the root navigator so that the route is pushed on top of the entire app. However, upon returning from the route, I can no longer interact with text fields. Pasting still works, but the keyboard does not appear.

This has been reproduced on both iOS and Android.

## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Interact with text field.
3. Click button to push new route.
4. Pop the route and return to the main screen.
5. Interact with the text field.

**Expected results:** Interacting with the text field causes the device keyboard to open.

**Actual results:** Interacting with the text field does not cause the device keyboard to open.

<details>
<summary>Code sample</summary>
I have included the code in a GitHub repository you can clone https://github.com/colehorvitz/flutter-keyboard-bug. Alternatively, I have pasted the code below.

```dart
import 'package:flutter/cupertino.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return CupertinoApp(
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return CupertinoTabScaffold(
        backgroundColor: CupertinoColors.white,
        tabBar: CupertinoTabBar(
          backgroundColor: CupertinoColors.white,
          items: [
            BottomNavigationBarItem(icon: Icon(CupertinoIcons.home)),
            BottomNavigationBarItem(icon: Icon(CupertinoIcons.search)),
            BottomNavigationBarItem(icon: Icon(CupertinoIcons.person)),
          ],
          inactiveColor: CupertinoColors.systemGrey2,
          activeColor: CupertinoColors.black,
        ),
        tabBuilder: (context, index) {
          return CupertinoTabView(
            builder: (context) {
              return MyPage(pageNumber: index);
            },
          );
        });
  }
}

class MyPage extends StatelessWidget {
  final int pageNumber;

  const MyPage({Key key, this.pageNumber}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      navigationBar: CupertinoNavigationBar(),
      backgroundColor: CupertinoColors.white,
      child: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
        Text(
          "Page $pageNumber",
          style: TextStyle(fontSize: 48),
        ),
        SizedBox(height: 32),
        Container(
            margin: EdgeInsets.symmetric(horizontal: 16),
            child: CupertinoTextField(
              placeholder: "Enter text",
              style: TextStyle(color: CupertinoColors.black, fontSize: 18),
              placeholderStyle:
                  TextStyle(color: CupertinoColors.systemGrey2, fontSize: 18),
              decoration: BoxDecoration(
                  color: CupertinoColors.white,
                  border: Border(
                      bottom: BorderSide(color: CupertinoColors.systemGrey3))),
            )),
        SizedBox(
          height: 16,
        ),
        CupertinoButton(
          color: CupertinoColors.activeBlue,
          onPressed: () {
            Navigator.of(context, rootNavigator: true).push(CupertinoPageRoute(
              builder: (context) {
                return CupertinoPageScaffold(
                  backgroundColor: CupertinoColors.white,
                  child: Center(
                    child: Text("Hello World"),
                  ),
                );
              },
            ));
          },
          child: Text(
            "Click Me",
            style: TextStyle(color: CupertinoColors.white),
          ),
        )
      ]),
    );
  }
}

```

</details>

<details>

```
flutter doctor
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.0.1 21A559 darwin-x64, locale en-US)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.64.2)
[✓] Connected device (1 available)
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @colehorvitz, you appear to be on an older version of flutter. Kindly upgrade to the latest version of stable (v 2.10.2) and let me know if you still experience this issue. 

If you do, please provide the updated `flutter doctor -v` and if possible, a screen recording that captures the issue accurately. 

Thank you

**Comment by colehorvitz** [comment]
Sorry @danagbemava-nc , the `flutter doctor` command I listed was incorrect. This was indeed tested with Flutter 2.8.1. I have updated the post and will follow up with a video shortly :)

**Comment by colehorvitz** [comment]
This is fixed on Flutter 2.10.2. Closing.
